src/core/components/object.js

```diff
@@ -210,14 +217,14 @@ module.exports = function object (self) {
         options = {}
       }
 
-      let mh
+      let mh, cid
 
       try {
         mh = normalizeMultihash(multihash, options.enc)
+        cid = new CID(mh)
       } catch (err) {
-        return callback(err)
+        return setImmediate(() => callback(errCode(err, 'ERR_INVALID_CID')))
```

Comment:
If `normalizeMultihash` throws the error users are going to get confused since this will always say the cid is the problem. We should check if it's an unsupported hash error first.